### ISSUE TITLE
Update for neato d8-d10

### ISSUE THREAD

**Issue by grantdray** [naofireblade/homebridge-neato#74]
Will this ever get upgraded to the newest models? @naofireblade

**Comment by Jason1976** [comment]
I also need the newest model D10, any update?

**Comment by naofireblade** [comment]
Do you get a specific error on the D8/D10? I don't have one for testing.

**Comment by Jason1976** [comment]
> Do you get a specific error on the D8/D10? I don't have one for testing.

Just show the below message.
[6/13/2022, 3:14:49 PM] [NeatoVacuumRobot] Successful login but no robots associated with your account.

I make sure that I have an D10 robotics and registed to MyNeato. Any idea ?

**Comment by naofireblade** [comment]
Do you see your robot here? https://help.neatorobotics.com/my-neato/

**Comment by Jason1976** [comment]
> Do you see your robot here? https://help.neatorobotics.com/my-neato/

No, I don't see any item. Am I need manually to register again ?

**Comment by Jason1976** [comment]
> > Do you see your robot here? https://help.neatorobotics.com/my-neato/
> 
> No, I don't see any item. Am I need manually to register again ?

I have contacted the neat support team and give me the below answer.
[This page](https://help.neatorobotics.com/my-neato/) is not used any longer, so you do not have to register your robot there.

**Comment by Jason1976** [comment]
> Do you see your robot here? https://help.neatorobotics.com/my-neato/

I just contacted the Neato support team and gave me the below answer.
[This page](https://help.neatorobotics.com/my-neato/) is not used any longer, so you do not have to register your robot there.


So this page didn't have newest model include D8, D9 and D10, any idea ?

**Comment by jiusanzhou** [comment]
The D8-D10 use a new app and API

**Comment by Jason1976** [comment]
> The D8-D10 use a new app and API

Yes, it’s corrected answer.

**Comment by grantdray** [comment]
Will someone update to make it work with new api and app?

**Comment by jiusanzhou** [comment]
Recently I'm trying rooting the D
9, I will post what I have done in few days later

**Comment by grantdray** [comment]
> Do you see your robot here? https://help.neatorobotics.com/my-neato/

Unfortunately, this isn't for the newer d8-d10 robots. this is for the older generation d7 and older. Is there anyway to update the api to get the robots from the new website/app so d8-d10 work with this plugin?

**Comment by grantdray** [comment]
> Do you see your robot here? https://help.neatorobotics.com/my-neato/

Any update on if you'll be able to update this for the new api and robots?

**Comment by RileyEv** [comment]
I've recently acquired a D9. After a bit of digging I've found most of the API endpoints needed to control it and am working on a version that should support the new style robots 😄

**Comment by grantdray** [comment]
Yay! I can’t wait. Let us know when it’s completed?!On Aug 25, 2022, at 6:06 PM, Riley Evans ***@***.***> wrote:﻿
I've recently acquired a D9. After a bit of digging I've found most of the API endpoints needed to control it and am working on a version that should support the new style robots 😄

**Comment by naofireblade** [comment]
> I've recently acquired a D9. After a bit of digging I've found most of the API endpoints needed to control it and am working on a version that should support the new style robots 😄

Awesome. Could you fork from the „dynamic-platform“ branch? That’s where I startet a complete rebuild of the plug-in to mitigate some problems.

**Comment by jamcycler** [comment]
Appreciate your work @RileyEv.

**Comment by grantdray** [comment]
@RileyEv How's the new version coming??

**Comment by BenjaminPaap** [comment]
I just recently wrote a python library to control a Neato D10 which is working good right now. If you guys need help feel free to contact me. I've got most of the new API endpoints in a JetBrains http client file. Should be easy to convert this into postman or something else. 

Here is the repo: https://github.com/BenjaminPaap/pyneato

**Comment by grantdray** [comment]
> I just recently wrote a python library to control a Neato D10 which is working good right now. If you guys need help feel free to contact me. I've got most of the new API endpoints in a JetBrains http client file. Should be easy to convert this into postman or something else. 
> 
> Here is the repo: https://github.com/BenjaminPaap/pyneato

Assuming we can use this on hoobs? I'll try it today :)

**Comment by grantdray** [comment]
> I just recently wrote a python library to control a Neato D10 which is working good right now. If you guys need help feel free to contact me. I've got most of the new API endpoints in a JetBrains http client file. Should be easy to convert this into postman or something else.
> 
> Here is the repo: https://github.com/BenjaminPaap/pyneato

@BenjaminPaap  is there a way to try this out with hoobs?

**Comment by BenjaminPaap** [comment]
Honestly I don't know much about hoobs. Personally I have a Homebridge instance running, but mostly switched to Home Assistant.  In my early days with homebridge I stumbled upon this issue and now just wanted to share my work in order to act as a blueprint. Maybe I'll write a JavaScript library too, but for the moment the python library is all I have for you.
